### PR TITLE
Initial pass at lint and test GHA for UI.

### PR DESCRIPTION
**Pull request by hatched**
Adds GHA's for testing and linting the UI.

### CODE REVIEW

File: .github/workflows/lint-ui.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - name: Checkout OSS Teleport
+        uses: actions/checkout@v3
+
+      - name: Setup Node Toolchain
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Install Yarn dependencies
+        run: yarn
```
Comment (zmb3) [inline .github/workflows/lint-ui.yaml:26]: `--frozen-lockfile`?

Comment (ravicious) [reply .github/workflows/lint-ui.yaml]: > `--frozen-lockfile`?

It doesn't make a difference when using workspaces with yarn v1, yarn v2 fixes this.

https://github.com/gravitational/webapps/blob/0fb98f4606f4d8b1854950f5b923afc7e42a662b/packages/build/scripts/yarn-install-frozen-lockfile.sh

**Review by ryanclark — APPROVED** [review]

**Review by jakule — APPROVED** [review]

File: web/packages/build/.eslintrc.js [hunk 2]
```diff
@@ -112,7 +112,8 @@ module.exports = {
     'import/no-named-as-default': 0,
     'import/default': 2,
     'import/named': 2,
-    'import/no-unresolved': 2,
+    // XXX Change to a 2 once e pkg imports are removed from teleterm.
```
Comment (zmb3) [inline web/packages/build/.eslintrc.js:115]:
```suggestion
    // TODO(hatched) Change to a 2 once e pkg imports are removed from teleterm.
```

**Review by zmb3 — APPROVED** [review]

**Comment by ravicious** [comment]
I double checked that ts-ignore statements and they seem like a good stop gap for now. They let tsc through and don't meaningfully change anything about the OSS build (it's still broken as those paths can't be imported). If you get e cloned everything works as usual, as long as those types don't get renamed (but TS lang server rename will work here) or removed.

idk how long until we manage to get rid of them, it might be two weeks, it might be a month.

**Pull request merged**